src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -140,27 +142,47 @@
   (log/info "Marking tables as inactive:"
             (for [table old-tables]
               (sync-util/name-for-logging (mi/instance Table table))))
-  (doseq [{schema :schema, table-name :name, :as _table} old-tables]
+  (doseq [{schema :schema table-name :name :as _table} old-tables]
     (t2/update! Table {:db_id  (u/the-id database)
                        :schema schema
                        :name   table-name
                        :active true}
                 {:active false})))
 
-(mu/defn ^:private update-table-description!
-  "Update description for any `changed-tables` belonging to `database`."
-  [database       :- i/DatabaseInstance
-   changed-tables :- [:set i/DatabaseMetadataTable]]
-  (log/info "Updating description for tables:"
-            (for [table changed-tables]
-              (sync-util/name-for-logging (mi/instance Table table))))
-  (doseq [{schema :schema, table-name :name, description :description} changed-tables]
-    (when-not (str/blank? description)
-      (t2/update! Table {:db_id       (u/the-id database)
-                         :schema      schema
-                         :name        table-name
-                         :description nil}
-                  {:description description}))))
+(mu/defn ^:private update-table-metadata-if-needed!
+  "Update the table metadata if it has changed."
+  [table-metadata :- i/DatabaseMetadataTable
+   metabase-table :- (ms/InstanceOf :model/Table)]
+  (log/infof "Updating table metadata for %s" (sync-util/name-for-logging metabase-table))
+  (let [to-update-keys [:description :database_require_filter]
+        old-table      (select-keys metabase-table to-update-keys)
+        new-table      (select-keys (merge
+                                     (zipmap to-update-keys (repeat nil))
+                                     table-metadata)
+                                    to-update-keys)
+        [_ changes _]  (data/diff old-table new-table)
+        changes        (cond-> changes
```

Comment:
this is a specical case, check test `metabase.sync.sync-metadata.comments-test/sync-existing-table-comment-test` and `metabase.sync.sync-metadata.comments-test/dont-overwrite-table-custom-description-test` to know how it's expected to work.